### PR TITLE
PolynomialDecayWithWarmup + tests

### PR DESCRIPTION
**Pull request by chaosagent**

### CODE REVIEW

File: examples/mlperf/lr_schedulers.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# https://github.com/mlcommons/training/blob/e237206991d10449d9675d95606459a3cb6c21ad/image_classification/tensorflow2/lars_util.py
+class PolynomialDecayWithWarmup(LR_Scheduler):
+  def __init__(self, optimizer: Optimizer, initial_lr, end_lr, train_steps, warmup, power=2):
+    super().__init__(optimizer)
+    assert train_steps > 0 and warmup > 0
+    self.warmup = min(warmup, train_steps)
+    self.initial_lr, self.end_lr, self.epochs, self.power = initial_lr, end_lr, train_steps, power
+
+    # set lr for first warmup step
+    self.optimizer.lr.assign(lr.cast(self.optimizer.lr.dtype) if isinstance(lr := self.get_lr(), Tensor) else lr).realize()
```
Comment (chenyuxyz) [inline examples/mlperf/lr_schedulers.py:15]: why would `self.get_lr` not be a Tensor? can you annotate output type for `get_lr`?

File: test/external/external_test_optim.py [hunk 2]
```diff
@@ -84,5 +111,13 @@ def test_multistep_lars(self): self._test_lamb(10, {'lr': 0.001}, 1e-5, 0)
   def test_multistep_lars_high_lr(self): self._test_lamb(10, {'lr': 10}, 1e-5, 3e-4)
   def test_lars_skip_list(self): self._test_lars(1, {'lr': 0.01, 'skip_list': True}, 1e-5, 0)
 
+  def test_lars_polylr(self):
+    self._test_lars_polylr(10, {'lr': 1.0}, {
```
Comment (chenyuxyz) [inline test/external/external_test_optim.py:115]: can you add one more with bigger steps? maybe the one that resnet will use

**Review by chenyuxyz — COMMENTED** [review]
looks good! minor comments

File: examples/mlperf/lr_schedulers.py (hunk 1 above)
Comment (chaosagent) [reply examples/mlperf/lr_schedulers.py:15]: unfortunately lr_schedule.py is really old. half of the schedulers don't work with jit, and some of them return floats. needs work.

File: test/external/external_test_optim.py (hunk 2 above)
Comment (chaosagent) [reply test/external/external_test_optim.py:115]: bigger on what dimension?

Comment (chenyuxyz) [reply test/external/external_test_optim.py:115]: sorry i meant steps, just one with different steps and warmup

File: examples/mlperf/lr_schedulers.py (hunk 1 above)
Comment (chaosagent) [reply examples/mlperf/lr_schedulers.py:15]: the reason why this line is so long is because resnet train_steps is larger than the fp16 largest int value, so we need epoch_counter in fp32 always, and thus we need to cast the output of get_lr. thought i would do a separate pr for that, though

Comment (chenyuxyz) [reply examples/mlperf/lr_schedulers.py:15]: i think break it into two lines with a TODO note helps, it's fine to fix other scheduler in a separate pr

File: test/external/external_test_optim.py (hunk 2 above)
Comment (chaosagent) [reply test/external/external_test_optim.py:120]: Hm, seems like it can run through 45 * 2053 steps pretty fast; added

File: examples/mlperf/lr_schedulers.py (hunk 1 above)
Comment (chaosagent) [reply examples/mlperf/lr_schedulers.py]: needs other changes in lr_scheduler.py anyways, will defer them until resnet fp16 PR
